### ISSUE TITLE
UseSqlServerStorage with connection string name does not work in .NET Core 3.1

### ISSUE THREAD

**Issue by kendaleiv** [HangfireIO/Hangfire#1590]
https://github.com/HangfireIO/Hangfire/blob/910795570011b95a9dbe6d2f284bebfd4d5a6f5e/src/Hangfire.SqlServer/SqlServerStorage.cs#L358-L376 does not use `ConfigurationManager` in the netstandard2.0 build. Specifying a connection string name, rather than a full connection string, results in:

```
An exception of type 'System.ArgumentException' occurred in System.Data.SqlClient.dll but was not handled in user code: 'Format of the initialization string does not conform to specification starting at index 0.'
```

**Comment by pieceofsummer** [comment]
There's no good way to resolve `IConfiguration` without introducing extra dependencies and adding a new overload for `UseSqlServerStorage()`.

As a workaround, you may use something like this:
```c#
services.AddHangfire((container, configuration) => configuration
     .UseSqlServerStorage(container.GetRequiredService<IConfiguration>().GetConnectionString("connectionStringName"))
);
```

**Comment by kendaleiv** [comment]
Yeah there's an easy workaround by providing the connection string. However, the parameter being named `nameOrConnectionString` isn't living up to its expectation in this case.
